### PR TITLE
Add dnf_config_manager module

### PR DESCRIPTION
**Pull request by ahyattdev**
##### SUMMARY
This enables Ansible to enable and disable repos with `dnf config-manager` using the repo IDs.

There has been community discussion surrounding this gap (the config-manager plugin won't be supported by the `ansible.builtin.dnf` module):
- https://github.com/ansible/ansible/issues/46963#issuecomment-1320976197
- https://www.reddit.com/r/ansible/comments/10muq0k/please_help_me_with_best_practices_in_my_first/

The Ansible [yum_repository](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_repository_module.html) module does not cover this use case since you are required to provide a repository URL. However, `dnf config-manager` only requires a repository ID to function.

##### ISSUE TYPE
- New Module/Plugin Pull Request

##### COMPONENT NAME
`dnf_config_manager`

##### ADDITIONAL INFORMATION

I believe that the `community.general` collection is a good fit for this module due to its inclusion of [dnf_versionlock](https://docs.ansible.com/ansible/latest/collections/community/general/dnf_versionlock_module.html) module, which also interfaces with a DNF plugin.

### CODE REVIEW

File: plugins/modules/dnf_config_manager.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: dnf_config_manager
+short_description: Enable or disable dnf repositories using config-manager
+version_added: 8.2.0
+description: This module enables or disables repositories using the C(dnf config-manager) sub-command.
```
Comment (felixfontein) [inline plugins/modules/dnf_config_manager.py:15]: The main description should make it easier to add more paragraphs to.
```suggestion
description:
  - This module enables or disables repositories using the C(dnf config-manager) sub-command.
```

File: plugins/modules/dnf_config_manager.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+author: Andrew Hyatt (@ahyattdev) <andy@hyatt.xyz>
+requirements:
+  - dnf
+  - dnf-plugins-core
+notes:
+  - Supports C(check_mode).
```
Comment (felixfontein) [inline plugins/modules/dnf_config_manager.py:21]: Please remove this, check mode is already documented with attributes.

File: plugins/modules/dnf_config_manager.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  name:
+    description:
+      - Repository id, for example C(crb).
```
Comment (felixfontein) [inline plugins/modules/dnf_config_manager.py:32]:
```suggestion
      - Repository ID, for example V(crb).
```
Also this option is a list, not a string, so the description seems off.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments. Please note that new plugins and modules must always come with tests.

**Comment by ahyattdev** [comment]
@felixfontein thanks for the feedback! I have added unit tests and worked in changes based on your comments.

File: plugins/modules/dnf_config_manager.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  name:
+    description:
+      - Repository id, for example V(crb).
```
Comment (felixfontein) [inline plugins/modules/dnf_config_manager.py:31]:
```suggestion
      - Repository ID, for example V(crb).
```

File: plugins/modules/dnf_config_manager.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+RETURN = r'''
+repo_states_pre:
+  description: Repo IDs before action taken.
+  returned: success
+  type: dict
+  contains:
+    repo_id:
+      description: Repository ID
+      type: str
+  sample: |-
```
Comment (felixfontein) [inline plugins/modules/dnf_config_manager.py:68]: The type is dict, not string, so you should pass a dict.
```suggestion
  sample:
```

File: plugins/modules/dnf_config_manager.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+'''
+
+RETURN = r'''
+repo_states_pre:
+  description: Repo IDs before action taken.
+  returned: success
+  type: dict
+  contains:
+    repo_id:
+      description: Repository ID
```
Comment (felixfontein) [inline plugins/modules/dnf_config_manager.py:66]:
```suggestion
      description: Repository ID.
```

File: plugins/modules/dnf_config_manager.py [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+      "crb-debug": "disabled",
+      "crb-source": "disabled"
+    }
+repo_states_post:
+  description: Repo IDs after action taken.
+  returned: success
+  type: dict
+  contains:
+    repo_id:
+      description: Repository ID
```
Comment (felixfontein) [inline plugins/modules/dnf_config_manager.py:86]:
```suggestion
      description: Repository ID.
```

File: plugins/modules/dnf_config_manager.py [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+      type: str
+  sample: |-
```
Comment (felixfontein) [inline plugins/modules/dnf_config_manager.py:88]:
```suggestion
  sample:
```

**Comment by ahyattdev** [comment]
@felixfontein thanks for the comments. Just pushed some changes updating the documentation and changing the return values.

The repository state return values now don't have varying keys. So the documentation on return value should be more clear now.

**Comment by felixfontein** [comment]
Looks good to me. If nobody objects I'll merge this in ~a week.

**Comment by felixfontein** [comment]
@ahyattdev thanks for your contribution!

**Comment by ahyattdev** [comment]
@felixfontein thanks for the review comments. This is a very organized project and I learned some more Python :slightly_smiling_face:

**Pull request merged**